packages/fluentui/react-builder/src/narration/NarrationComputer.tsx

```diff
@@ -23,14 +35,186 @@ export class NarrationComputer {
-  async compute(element: Element, platform: string): Promise<string> {
+  // Traverses the DOM rooted at the given element to find all the focusable elements and returns them.
+  async getFocusableElements(element: IAriaElement): Promise<FocusableElement[]> {
+    // Prepare all the arrays
+    const activeDescendantsParents: IAriaElement[] = [];
+    const path: string[] = [];
+    const focusableElements: FocusableElement[] = [];
+
+    // Traverse down the DOM tree rooted at the element to first find the activeDescendants parent elements, and then all the focusable elements and their paths
+    this.findActiveDescendantsParents(element, activeDescendantsParents);
+    await this.findFocusableElements(element, path, focusableElements, activeDescendantsParents);
+
+    return focusableElements;
+  } // End getFocusableElements
+
+  // Recursively traverses the DOM tree rooted at the given element to find all the parents which have the ariaActiveDescendantElement property set, and saves the found elements to the given parents array.
+  findActiveDescendantsParents(element: IAriaElement, parents: IAriaElement[]) {
+    if (element.ariaActiveDescendantElement != null) {
+      // Begin if 1
+      parents.push(element);
+    } // End if 1
+    for (let i = 0; i < element.children.length; i++) {
+      // Begin for 1
+      const child = element.children[i] as IAriaElement;
+      this.findActiveDescendantsParents(child, parents);
+    } // End for 1
+  } // End findActiveDescendantsParents
+
+  // Recursively traverses the given element to find all the focusable elements and saves their paths along the way.
+  async findFocusableElements(
+    element: IAriaElement,
+    path: string[],
+    focusableElements: FocusableElement[],
+    activeDescendantsParents: IAriaElement[],
+  ) {
+    // Determine the path
+    const newPath = path.slice();
+    const node = await (window as any).getComputedAccessibleNode(element);
+
+    // Only add the item to the path if the role is not a generic container
+    if (node.role !== 'genericContainer') {
+      const item = node.name ? `${node.role} (${node.name})` : node.role;
+      newPath.push(item);
+    }
+
+    // If the element is focusable, save it together with its path
+    const isDirectlyFocusable =
+      (element.getAttribute('tabindex') || element.tabIndex >= 0) && element.ariaActiveDescendantElement == null;
+    const isActiveDescendant = activeDescendantsParents.some(parent => {
+      // Begin some 1
+      return parent.tabIndex >= 0 && parent.ariaActiveDescendantElement === element;
+    }); // End some 1
+    if (isDirectlyFocusable || isActiveDescendant) {
+      // Begin if 1
+      const focusableElement: FocusableElement = {
+        path: newPath,
+        element,
+      };
+      focusableElements.push(focusableElement);
+      // return;
+    } // End if 1
+
+    // Traverse down to all the children
+    for (let i = 0; i < element.children.length; i++) {
```

Comment:
`.forEach`?

Reply:
It's an HTMLCollection array-like object which cannot be iterated using forEach().

Reply:
Tried to replace it with Array.from(element.children).forEach(async (child: IAriaElement) => {...}); Although it does not complain at compile time, it does not work in runtime. Giving me some errors wihch seems unrelated to this problem.

Reply:
After a Teams meeting, we decided to keep it using the for statement.